src/Components/Components/src/PersistentState/PersistentValueProviderComponentSubscription.cs

```diff
@@ -233,10 +233,28 @@ private static PropertyGetter PropertyGetterFactory((Type type, string propertyN
         {
             throw new InvalidOperationException($"Property {propertyName} not found on type {type.FullName}");
         }
+
+        // Check if the property is public
+        if (propertyInfo.GetMethod == null || !propertyInfo.GetMethod.IsPublic)
+        {
+            throw new InvalidOperationException(
+                $"The property '{propertyName}' on component type '{type.FullName}' cannot be used with PersistentState because it is not public. Properties with PersistentState must be public.");
```

Comment:
Use `nameof(PersistentStateAttribute)` in the error message. Mention that the property must have a public getter (Or whatever similar language we use in ComponentProperties.cs)